### PR TITLE
Feature/fga/authenticated jitsi

### PR DESCRIPTION
**Pull request by ganfra**
PR to support authenticated jitsi:  https://github.com/vector-im/element-android/issues/3379

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/thirdparty/ThirdPartyAPI.kt [hunk 1]
```diff
@@ -41,4 +45,16 @@ internal interface ThirdPartyAPI {
...
+
+    /**
+     * Gets an OpenID token object that the requester may supply to another service to verify their identity in Matrix.
+     * The generated token is only valid for exchanging for user information from the federation API for OpenID.
+     * The access token generated is only valid for the OpenID API. It cannot be used to request another OpenID access token or call /sync, for example.
+     *
+     * Ref: https://matrix.org/docs/spec/client_server/latest#post-matrix-client-r0-user-userid-openid-request-token
+     */
+    @POST(NetworkConstants.URI_API_PREFIX_PATH_R0 + "user/{userId}/openid/request_token")
+    suspend fun requestOpenIdToken(@Path("userId") userId: String,
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/thirdparty/ThirdPartyAPI.kt:57]: Why not reusing `OpenIdAPI`?

File: vector/src/main/java/im/vector/app/core/utils/Base32.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+package im.vector.app.core.utils
+
+import org.apache.commons.codec.binary.Base32
+
+fun String.toBase32String(padding: Boolean = true): String {
+    val base32 = Base32().encodeAsString(toByteArray())
+    return if (padding) {
+        base32
+    } else {
+        base32.replace("=", "")
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/utils/Base32.kt:25]: use `trim()` instead?

File: vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt [hunk 3]
```diff
@@ -38,6 +38,13 @@ internal fun String.ensureProtocol(): String {
     }
 }
 
+/**
+ * Ensure string do not starts with "http" or "https" protocol.
+ */
+internal fun String.ensureNoProtocol(): String {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt:44]: `internal` is a bit useless here

File: vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetPropertiesFactory.kt [hunk 4]
```diff
@@ -37,7 +37,7 @@ class JitsiWidgetPropertiesFactory @Inject constructor(
                 .orEmpty()
 
         return JitsiWidgetProperties(
-                domain = configs["conferenceDomain"] ?: stringProvider.getString(R.string.preferred_jitsi_domain),
+                domain = (configs["conferenceDomain"] ?: stringProvider.getString(R.string.preferred_jitsi_domain)),
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetPropertiesFactory.kt:40]: Why adding those extra parenthesis?

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 5]
```diff
@@ -267,67 +263,67 @@ class RoomDetailViewModel @AssistedInject constructor(
...
                 _viewEvents.post(
                         RoomDetailViewEvents.ShowRoomAvatarFullScreen(action.matrixItem, action.transitionView)
                 )
             }
-            is RoomDetailAction.DoNotShowPreviewUrlFor -> handleDoNotShowPreviewUrlFor(action)
-            RoomDetailAction.RemoveAllFailedMessages -> handleRemoveAllFailedMessages()
-            RoomDetailAction.ResendAll -> handleResendAll()
+            is RoomDetailAction.DoNotShowPreviewUrlFor           -> handleDoNotShowPreviewUrlFor(action)
+            RoomDetailAction.RemoveAllFailedMessages             -> handleRemoveAllFailedMessages()
+            RoomDetailAction.ResendAll                           -> handleResendAll()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:326]: @onurays will not be happy that this is changing again...

**Review by bmarty — COMMENTED** [review]
Some quick remarks after a first review

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/thirdparty/ThirdPartyAPI.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/thirdparty/ThirdPartyAPI.kt:57]: Woah, I didn't see this one

File: vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetPropertiesFactory.kt (hunk 4 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetPropertiesFactory.kt:40]: it's a mistake

File: vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt (hunk 3 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt:44]: Used same qualifier than other methods in the file, but I guess it's some copy/paste from SDK, will change

File: vector/src/main/java/im/vector/app/core/utils/Base32.kt (hunk 2 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/core/utils/Base32.kt]: done

File: vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt (hunk 3 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/core/utils/UrlUtils.kt]: Finally removed the method, I do not need it anymore. I still removed the internal keywords from the file

**Review by bmarty — APPROVED** [review]
Thanks for the update, LGTM.
How can I test a Jitsi Widget which requires auth?

**Pull request merged**